### PR TITLE
Enable attaching exception record to a case by legacy ID

### PR DESCRIPTION
**Pull request by lgonczar**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java [hunk 1]
```diff
@@ -107,7 +110,7 @@ void should_attach_exception_record_to_the_existing_case_with_evidence_documents
         CaseDetails exceptionRecord = createExceptionRecord("envelopes/supplementary-evidence-envelope.json");
 
         // when
-        invokeCallbackEndpointForLinkingDocsToCase(caseDetails, exceptionRecord);
+        invokeCallbackEndpoint(caseDetails, exceptionRecord, null);
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: what does null `searchCaseReferenceType` mean?

Comment (lgonczar) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: It means no searchCaseReferenceType is provided. I'll add a comment.

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java [hunk 2]
```diff
@@ -136,7 +192,32 @@ private void invokeCallbackEndpointForLinkingDocsToCase(CaseDetails caseDetails,
...
+        String searchCaseReferenceType
+    ) {
+        Map<String, Object> exceptionRecordData = new HashMap<>(exceptionRecord.getData());
+
+        if (searchCaseReferenceType == null) {
+            exceptionRecordData.put("attachToCaseReference", String.valueOf(targetCaseDetails.getId()));
+        } else {
+            exceptionRecordData.put("searchCaseReferenceType", searchCaseReferenceType);
+
+            String searchCaseReference = EXTERNAL_CASE_REFERENCE.equals(searchCaseReferenceType)
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: `searchCaseReferenceType` is not null, so it's safe to compare variable to a constant the normal way.
Alternatively, there's always `Objects.equals(a, b)`

Comment (lgonczar) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: Alternatives are just as good as current version. Is it the order that bothers you?

Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: to me it looks like `if (5 == x)` which I personally find quite weird

Comment (lgonczar) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java]: `5 == x` would be weird, but `"CONSTANT".equals(var)` is pretty conventional. You're right it's not needed here, so I changed it.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 3]
```diff
@@ -42,67 +51,153 @@ public AttachCaseCallbackService(CcdApi ccdApi) {
...
     }
 
     //The code below need to be rewritten to reuse the EventPublisher class
 
-    private List<String> attachToCase(
-        String exceptionRecordJurisdiction,
-        String targetCaseRef,
-        Long exceptionRecordId,
-        List<Map<String, Object>> exceptionRecordDocuments
+    private Either<List<String>, Map<String, Object>> tryAttachToCase(
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: comment explaining what's in the 'either' object would be nice

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: I've got it on the main method. Can add it here, too.

**Pull request merged**